crates/iceberg/src/spec/values.rs

```diff
@@ -2682,12 +2682,79 @@ mod _serde {
                     Type::Primitive(PrimitiveType::String) => Ok(Some(Literal::string(v))),
                     _ => Err(invalid_err("string")),
                 },
-                // # TODO:https://github.com/apache/iceberg-rust/issues/86
-                // rust avro don't support deserialize any bytes representation now.
-                RawLiteralEnum::Bytes(_) => Err(invalid_err_with_reason(
-                    "bytes",
-                    "todo: rust avro doesn't support deserialize any bytes representation now",
-                )),
+                RawLiteralEnum::Bytes(v) => match ty {
+                    Type::Primitive(PrimitiveType::Binary) => Ok(Some(Literal::binary(v.to_vec()))),
+                    Type::Primitive(PrimitiveType::Fixed(expected_len)) => {
+                        if v.len() == *expected_len as usize {
+                            Ok(Some(Literal::fixed(v.to_vec())))
+                        } else {
+                            Err(invalid_err_with_reason(
+                                "bytes",
+                                &format!(
+                                    "Fixed type must be exactly {} bytes, got {}",
+                                    expected_len,
+                                    v.len()
+                                ),
+                            ))
+                        }
+                    }
+                    Type::Primitive(PrimitiveType::Uuid) => {
+                        if v.len() == 16 {
+                            let bytes: [u8; 16] = v.as_slice().try_into().map_err(|_| {
+                                invalid_err_with_reason("bytes", "UUID must be exactly 16 bytes")
+                            })?;
+                            Ok(Some(Literal::Primitive(PrimitiveLiteral::UInt128(
```

Comment:
This is just a minor nit, but is this equivalent to
`Ok(Some(Literal::uuid(uuid::Uuid::from_bytes(bytes))))`
https://github.com/Fokko/iceberg-rust/blob/2b31cb4e267698817b90e92854fa971980f97dde/crates/iceberg/src/spec/values.rs#L2848
further down in the file? If so, should we be consistent?